File: frontend/src/types/task.ts
// Типы для модуля задач

export type TaskCategory = 'WORK' | 'STUDY' | 'VOLUNTEER' | 'DELIVERY' | 'SERVICES' | 'OTHER';

export type TaskStatus = 'DRAFT' | 'OPEN' | 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED' | 'EXPIRED';

export interface Task {
  id: number;
  title: string;
  description: string;
  category: TaskCategory;
  creditReward: number;
  status: TaskStatus;
  creatorId: number;
  executorId?: number;
  location?: {
    latitude: number;
    longitude: number;
    address?: string;
    city?: string;
  };
  deadline?: string;
  createdAt: string;
  updatedAt: string;
}

export interface TaskCreateRequest {
  title: string;
  description: string;
  category: TaskCategory;
  creditReward: number;
  deadline?: string;
  latitude?: number;
  longitude?: number;
  address?: string;
}

export interface TaskSearchRequest {
  lat?: number;
  lon?: number;
  radiuskm?: number;
  category?: TaskCategory;
  minCredit?: number;
  maxCredit?: number;
  status?: TaskStatus;
  deadlineBefore?: string;
  page?: number;
  size?: number;
}

export interface Bid {
  id: number;
  taskId: number;
  userId: number;
  message: string;
  proposedDeadline?: string;
  portfolioUrl?: string;
  status: 'ACTIVE' | 'ACCEPTED' | 'REJECTED';
  createdAt: string;
}

export interface BidCreateRequest {
  message: string;
  proposedDeadline?: string;
  portfolioUrl?: string;
}

File: task/src/main/java/com/trust/task/entity/Task.java
package com.trust.task.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Task {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable=false, length=200)
    private String title;

    @Column(nullable=false, length=2000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable=false)
    private TaskCategory category;

    @Column(nullable=false)
    private Integer creditReward;

    private LocalDateTime deadline;

    @Column(nullable = false)
    private Long creatorId;

    @Enumerated(EnumType.STRING)
    @Column(nullable=false)
    private TaskStatus status = TaskStatus.OPEN;

    @Column(nullable = true)
    private Long executorId;

    // ... дополнительные поля и методы

    // getters/setters
}
